apollo-federation/src/connectors/models/http_json_transport.rs

```diff
@@ -160,11 +187,12 @@ impl HttpJsonTransport {
             };
         }
         if let Some(connect_path) = self.connect_path.as_ref() {
-            warnings.extend(extend_path_from_expression(
-                &mut path,
-                connect_path,
-                inputs,
-            )?);
+            warnings.extend(
+                extend_path_from_expression(&mut path, connect_path, inputs)?
+                    .iter()
+                    .cloned()
```

Comment:
```suggestion
                    .into_iter()
```
Avoids some of the clones you were worried about 😁